dockers/base-conda/Dockerfile

```diff
@@ -111,18 +114,24 @@ RUN \
     rm requirements*
 
 RUN \
+    CUDA_VERSION_MAJOR=$(python -c "import torch ; print(torch.version.cuda.split('.')[0])") && \
+    py_ver=$(python -c "print(int('$PYTHON_VERSION'.split('.') >= '3.9'.split('.')))") && \
     # install DALI, needed for examples
-    pip install --extra-index-url https://developer.download.nvidia.com/compute/redist nvidia-dali-cuda${CUDA_VERSION%%.*}0
+    # todo: waiting for 1.4 - https://github.com/NVIDIA/DALI/issues/3144#issuecomment-877386691
+    if [ $py_ver -eq "0" ]; then \
+        pip install --extra-index-url https://developer.download.nvidia.com/compute/redist "nvidia-dali-cuda${CUDA_VERSION_MAJOR}0>1.0" ; \
+        python -c 'from nvidia.dali.pipeline import Pipeline' ; \
+    fi
 
 RUN \
     # install NVIDIA apex
-    # TODO: later commits break CI when cpp extensions are compiling. Unset when fixed
-    pip install --no-cache-dir --global-option="--cpp_ext" --global-option="--cuda_ext" git+https://github.com/NVIDIA/apex@705cba9
+    pip install --no-cache-dir --global-option="--cuda_ext" https://github.com/NVIDIA/apex/archive/refs/heads/master.zip && \
```

Comment:
You sure we want to point to master for APEX? just hesitant as I'd prefer stability over latest changes on the deprecated package

Reply:
at least actual master is fine, we can freeze it later if needed :]